test/e2e/lib/lifecycle.go

```diff
@@ -205,36 +206,40 @@ func (c *Client) LogsFor(namespace, name string, tm *metav1.TypeMeta) (string, e
 }
 
 // TODO make this function more generic.
-func (c *Client) StackDriverEventCountMetricFor(namespace, projectID, filter string) (*int64, error) {
-	metricClient, err := metrics.NewStackDriverMetricClient()
+func (c *Client) StackDriverEventCountMetricFor(namespace, projectID, filter string) (int64, error) {
+	metricClient, err := monitoring.NewMetricClient(context.TODO())
 	if err != nil {
-		return nil, fmt.Errorf("failed to create stackdriver metric client: %v", err)
+		return 0, fmt.Errorf("failed to create stackdriver metric client: %v", err)
 	}
 
 	// TODO make times configurable if needed.
-	metricRequest := metrics.NewStackDriverListTimeSeriesRequest(projectID,
-		metrics.WithStackDriverFilter(filter),
-		// Starting 5 minutes back until now.
-		metrics.WithStackDriverInterval(time.Now().Add(-5*time.Minute).Unix(), time.Now().Unix()),
+	metricRequest := &monitoringpb.ListTimeSeriesRequest{
+		Name:   fmt.Sprintf("projects/%s", projectID),
+		Filter: filter,
+		Interval: &monitoringpb.TimeInterval{
+			// Starting 5 minutes back until now.
+			StartTime: &timestamp.Timestamp{Seconds: time.Now().Add(-5 * time.Minute).Unix()},
+			EndTime:   &timestamp.Timestamp{Seconds: time.Now().Unix()},
+		},
 		// Delta counts aggregated every 2 minutes.
 		// We aggregate for count as other aggregations will give higher values.
 		// The reason is that PubSub upon an error, will retry, thus we will be recording multiple events.
-		metrics.WithStackDriverAlignmentPeriod(2*int64(time.Minute.Seconds())),
-		metrics.WithStackDriverPerSeriesAligner(monitoringpb.Aggregation_ALIGN_DELTA),
-		metrics.WithStackDriverCrossSeriesReducer(monitoringpb.Aggregation_REDUCE_COUNT),
-	)
-
-	it := metricClient.ListTimeSeries(context.TODO(), metricRequest)
+		Aggregation: &monitoringpb.Aggregation{
+			AlignmentPeriod:    &duration.Duration{Seconds: 120},
+			PerSeriesAligner:   monitoringpb.Aggregation_ALIGN_DELTA,
```

Comment:
I just copied the previous logic. What would be the correct aligner to use?

Reply:
Actually this is valid for cumulative metrics, it will convert the metric to a DELTA. The cross series aligner seems incorrect however, it will only count data points and disregard the metric values. We probably want REDUCE_SUM or REDUCE_NONE here. The filter may be changed to filter unwanted metrics based on the previous comment, however.

Reply:
Thanks for the insights! Since this is orthogonal to this change, how about addressing this in a separate PR (along with the multiple points issue below)? I am happy to send a follow up right after this one gets merged.

Reply:
Sounds good!

Reply:
Upon a second look, there is a comment above that says 
```
// Delta counts aggregated every 2 minutes.
// We aggregate for count as other aggregations will give higher values.
// The reason is that PubSub upon an error, will retry, thus we will be recording multiple events.
```

So this seems intentional. I created a PR to fix the multiple data points in #1510

Reply:
It may be intentional but the number of data points is not the correct thing to measure. If pubsub errors are an issue they should be handled in another way such as with a stricter filter.